File: oop/python/lista01/Ex04.py
from KeyboardInput import KeyboardInput

class Ex04:
    def printTree(self, height) -> None:
        currentHeight = height
        while currentHeight > 0:
            print(" " * (height - currentHeight) + "*" * currentHeight)
            currentHeight -= 1

if __name__ == "__main__":
    inverseTreePrinter = Ex04()
    keyboardInput = KeyboardInput()

    treeHeight = keyboardInput.readPositiveInt("Provide the inverse tree height: ")
    inverseTreePrinter.printTree(treeHeight)

File: oop/java/src/Lista01/Ex04/Ex04.java
package Lista01.Ex04;

import Lista01.KeyboardInput;

public class Ex04 {
    public void printTree(Integer height) {
        for (int currentHeight = height; currentHeight > 0; currentHeight--)
            System.out.println(
                " ".repeat(height - currentHeight) + "*".repeat(currentHeight)
            );
    }

    public static void main(String[] args) {
        Ex04 inverseTreePrinter = new Ex04();
        KeyboardInput keyboardInput = new KeyboardInput(System.in);

        System.out.println("Provide the inverse tree height: ");
        Integer input = keyboardInput.readPositiveInteger();

        inverseTreePrinter.printTree(input);
    }
}
